rfcs/0013-avalanche.md

```diff
@@ -0,0 +1,98 @@
+# Replication via Avalanche 
+
+This RFC describes the current single layer replication mechanism as well as proposed changes to add a second layer and implementing an Avalanche mechanism.
+
+## Current Design
+There's two basic parts to the current Replication design. 
+
+#### Broadcast stage:
+In this stage, the leader distributes its data across the Layer-1 nodes. Currently, Layer-1 nodes are all known "tvu peers" (ClusterInfo::tvu_peers). 
+That way each Layer-1 node only receives partial data from the leader and the Replicate Stage in each Layer-1 node's TVU will ensure all data is shared between its Layer-1 peers 
+and a complete window is received.   
+
+The broadcast stage sends its data and "broadcast_table" (tvu peers currently) to ClusterInfo to complete the broadcast. ClusterInfo then 
+sets up the broadcast "orders" by distributing the window's blobs across all broadcast_table entities. 
+
+#### Replicate stage:  
+The Replicate stage *forwards* data from a Layer-1 node to all of its "tvu peers". So once as nodes start seeing complete windows they can sending their votes back to the leader and once 2/3 + 1 majority is 
+reached we can say that finality has been achieved and the leader can move on.
+
+**Cluster_info -> retransmit** = Used by TVUs to retransmit. Currently Layer-1 sends this to all tvu peers, leader is automatically excluded. This stage has bank and should be able to use it for weighted selection in the new design.
+
+**Cluster_info -> broadcast** = Used by leader to broadcast to layer-1 nodes. 
+
+**BroadcastStage -> run** = Used by leader (TPU) to broadcast to all validators. Currently all TVU Peers are considered layer-1. But blobs are transmitted sort of round robin. See Cluster_info->Broadcast. This stage doesn't have bank and will need it for weighted selection.
+
+
+## New Design
+
+                                     .-------------.
```

Comment:
You'll want to put triple-backticks at the start and end of this so that GitHub markdown doesn't try to render it.

Reply:
Nevermind, I see you did the multiple leading spaces trick!